### DISCUSSION TITLE
Reduce RAM / Memory usage

### DISCUSSION THREAD

**Discussion by ngosang** [searxng/searxng#1892] [Q&A]
I'm self-hosting a SearXNG instance for personal use. It's deployed with Docker in a server with 4 cores.
It work really well but it uses 600 MB of RAM. Since I'm the only user and it's idle most of the time I did some changes in the configuration.

* Changing `workers = %k => workers = 1` in `uwsgi.ini` makes a huge difference. 600 MB => 150 MB

Do you know how to reduce the RAM usage even more?

**Comment by unixfox** [comment]
You can also not use uwsgi and run searx standalone using `python -m searx.webapp`, that should probably reduce the memory usage.

**Reply by ngosang** [reply]
I tried and and the memory usage is really close to using uwsgi with `workers = 1`.

**Comment by adriadam10** [comment]
If anyone doesn't know how to change the workers variable, it could be change by passing this argument in the docker compose:

`UWSGI_WORKERS=x`

**Comment by return42** [comment]
FYI: we started a patch series to reduce the memory footprint ...

- https://github.com/searxng/searxng/pull/4834
- https://github.com/searxng/searxng/pull/4836
- .. more will follow ..
